src/main/java/org/broadinstitute/hellbender/tools/exome/germlinehmm/IntegerCopyNumberTransitionMatrixCollectionReader.java

```diff
@@ -0,0 +1,118 @@
+ * table is formatted as follows:
+ *
+ *      CONTIG_NAME SEX_XX  SEX_XY
+ *      1   t_matrix_table_XX_1.tsv  t_matrix_table_XY_1.tsv
+ *      2   t_matrix_table_XX_2.tsv  t_matrix_table_XY_2.tsv
+ *      3   t_matrix_table_XX_3.tsv  t_matrix_table_XY_3.tsv
+ *                           . . .
+ *      X   t_matrix_table_XX_X.tsv  t_matrix_table_XY_X.tsv
+ *      Y   t_matrix_table_XX_Y.tsv  t_matrix_table_XY_Y.tsv
+ *
+ * The T-matrix table files ("t_matrix_table_[...].tsv") are expected to reside in the same directory as the
+ * collection table file. These files do not need to be unique, e.g. one can use the same copy number prior
+ * table for all autosomal contigs.
+ *
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public final class IntegerCopyNumberTransitionMatrixCollectionReader {
+
+    private static final String CONTIG_NAME_TABLE_COLUMN = "CONTIG_NAME";
+    private static final TableColumnCollection MANDATORY_COLUMNS = new TableColumnCollection(CONTIG_NAME_TABLE_COLUMN);
+
+    /**
+     * Reads the copy number transition prior table file and returns an instance of
+     * {@link IntegerCopyNumberTransitionMatrixCollection}.
+     *
+     * @param tableFile input table
+     * @return an instance of {@link IntegerCopyNumberTransitionMatrixCollection}
+     */
+    public static IntegerCopyNumberTransitionMatrixCollection read(@Nonnull final File tableFile) {
+        try {
+            return read(new FileReader(tableFile), tableFile.getParent());
+        } catch (final IOException ex) {
+            throw new UserException.CouldNotReadInputFile("Could not read the copy number transition matrix collection" +
+                    " table file: " + tableFile.getAbsolutePath());
+        }
+    }
+
+    /**
+     * Reads the copy number transition matrix collection from a reader and returns an instance of
+     * {@link IntegerCopyNumberTransitionMatrixCollection}.
+     *
+     * @param reader a reader instance for the collection
+     * @param parentPath the parent path for the transition matrices in the collection
+     * @return an instance of {@link IntegerCopyNumberTransitionMatrixCollection}
+     */
+    public static IntegerCopyNumberTransitionMatrixCollection read(@Nonnull final Reader reader,
+                                                                   @Nonnull final String parentPath) {
+        Utils.nonNull(reader, "The integer copy number transition matrix collection table reader must be non-null");
+        Utils.nonNull(parentPath, "The parent path of the integer copy number transition matrix collection table file" +
+                " must be non-null");
+        final IntegerCopyNumberTransitionMatrixCollection collection = new IntegerCopyNumberTransitionMatrixCollection();
+        try (final TableReader<TransitionMatrixCollectionRow> tableReader = new TableReader<TransitionMatrixCollectionRow>(reader) {
```

Comment:
I thought about that too but the bi-function exception factory makes it look more mysterious...